rmf_inorbit_fleet_adapter/rmf_inorbit_fleet_adapter/RobotCommandHandle.py

```diff
@@ -465,8 +467,18 @@ def get_remaining_waypoints(self, waypoints: list):
         may be modified if waypoints in a path need to be filtered.
         '''
         assert (len(waypoints) > 0)
-        remaining_waypoints = []
 
+        # When the "responsive wait" feature is enabled, RMF will send two consecutive identical
+        # waypoints with a 30" timestamp difference.
+        # The following lines filter the first waypoint of these pairs to prevent an excess of
+        # navigation requests from the fleet adapter
+        for i in range(1, len(waypoints)):
```

Comment:
Can't you index waypoints with i being out of bounds after popping an element?

Reply:
I am not sure what you mean by that. Could you please explain a bit more?

Reply:
Please try the following code adaptation in a python 3 interpreter:

```
class Waypoint:
    def __init__(self, x, y, time):
        self._position = [x, y]
        self._time = time
    
    @property
    def position(self):
        return self._position

    @property
    def time(self):
        return self._time
    
waypoints = [Waypoint(1, 2, 10), Waypoint(1, 2, 40), Waypoint(2, 3, 70)]

print("......")
for w in waypoints: print(w.position, w.time)
print("......")
for i in range(1, len(waypoints)):
    print(f"Index: {i} and {i-1}. len(waypoints): {len(waypoints)}")
    if waypoints[i-1].position == waypoints[i].position:
        print(f"Found duplicated consecutive waypoints in waypoints list: [{waypoints[i].position}, {waypoints[i].time}], [{waypoints[i-1].position}, {waypoints[i-1].time}]")
        waypoints.pop(i)
        i -= 1
    else:
        print("not duplicated")
print("......")
for w in waypoints: print(w.position, w.time)
```

I can see the following output:

```
......
[1, 2] 10
[1, 2] 40
[2, 3] 70
......
Index: 1 and 0. len(waypoints): 3
Found duplicated consecutive waypoints in waypoints list: [[1, 2], 40], [[1, 2], 10]
Index: 2 and 1. len(waypoints): 2
Traceback (most recent call last):
  File "/home/agalbachicar/test.py", line 24, in <module>
    if waypoints[i-1].position == waypoints[i].position:
IndexError: list index out of range

```

Reply:
I may have understood something wrong but the fact that you are iterating on waypoints original length and popping elements in the loop is not a good idea. You may find this [link](https://stackoverflow.com/questions/1207406/how-to-remove-items-from-a-list-while-iterating) useful.

Reply:
I understand what you mean now, thanks for taking the time to explain it. You are right, there is an error there. If you can, take a look now.